core/src/main/java/org/hibernate/ogm/util/impl/TupleContextHelper.java

```diff
@@ -6,29 +6,103 @@
  */
 package org.hibernate.ogm.util.impl;
 
+import java.util.List;
+import java.util.Map;
+
 import org.hibernate.engine.spi.SessionImplementor;
+import org.hibernate.ogm.dialect.batch.spi.OperationsQueue;
+import org.hibernate.ogm.dialect.spi.TransactionContext;
 import org.hibernate.ogm.dialect.spi.TupleContext;
+import org.hibernate.ogm.model.key.spi.AssociatedEntityKeyMetadata;
 import org.hibernate.ogm.model.spi.EntityMetadataInformation;
+import org.hibernate.ogm.options.spi.OptionsContext;
 import org.hibernate.ogm.persister.impl.OgmEntityPersister;
 
 /**
  * @author Davide D'Alto
  */
 public class TupleContextHelper {
 
+	private static final Log LOG = LoggerFactory.make();
+
 	/**
 	 * Given a {@link SessionImplementor} returns the {@link TupleContext} associated to an entity.
 	 *
 	 * @param session the current session
 	 * @param metadata the {@link EntityMetadataInformation} of the entity associated to the TupleContext
-	 * @return the TupleContext associated to the current session for the entity specified, or {@code null} if the
-	 * EntityMetadataInformation is {@code null}
+	 * @return the TupleContext associated to the current session for the entity specified
 	 */
 	public static TupleContext tupleContext(SessionImplementor session, EntityMetadataInformation metadata) {
 		if ( metadata != null ) {
 			OgmEntityPersister persister = (OgmEntityPersister) session.getFactory().getEntityPersister( metadata.getTypeName() );
 			return persister.getTupleContext( session );
 		}
-		return null;
+		else if ( session != null ) {
+			// We are not dealing with a single entity but we might still need the transactionContext
+			TransactionContext transactionContext = TransactionContextHelper.transactionContext( session );
+			TupleContext tupleContext = new OnlyWithTransactionContext( transactionContext );
+			return tupleContext;
+		}
+		else {
+			return null;
+		}
+	}
+
+	/**
+	 * This {@link TupleContext} can be used for those use case where we don't necessarily have a tuple context but we
+	 * still need a {@link TransactionContext}.
+	 */
+	private static class OnlyWithTransactionContext implements TupleContext {
```

Comment:
I understand the point you exposed in your PR comment as the PR is already big enough. But I don't like this.

I think we should try to get rid of this once we got this PR merged. @gunnarmorling any opinion about this?